configure.ac

```diff
@@ -126,14 +126,20 @@ AC_ARG_ENABLE([wallet],
   [AS_HELP_STRING([--disable-wallet],
   [disable wallet (enabled by default)])],
   [enable_wallet=$enableval],
-  [enable_wallet=yes])
+  [enable_wallet=auto])
```

Comment:
In commit "Allow disabling BDB in configure with --without-bdb" (26102beed00e51b3bb1e263287028d3e4b9d85c5)

Is this change intentional? Commit description only says that this is adding a new option, not that this is changing default build behavior and automatically building without legacy wallet support if BDB is not found. Similarly, I'd also expect `use_bdb` value below to be default `yes` instead of default `auto` to continue requiring BDB, unless user explicitly requests to build without it, or requests to build without the wallet.

Ignore this comment if I'm misinterpreting and there is no change in default behavior, but otherwise I think it'd be better to just either restore previous behavior and require BDB unless explicitly disabled, or document somewhere (commit message or release notes) that bitcoin by default may be now built without BDB support or any wallet support if libraries aren't detected.

Reply:
This doesn't change any behavior. I needed to do this to tell when the user explicitly does `--enable-wallet`. In that case, an error needs to be given when combined with `--without-bdb --without-sqlite`. Previously, `enable_wallet` would always be `yes` so it wasn't possible to automatically do `enable_wallet=no` when `--without-bdb --without-sqlite` without potentially confusing the user (it would then disable the wallet if they had done `--enable-wallet`).

Reply:
Are you saying before this change with default `./configure` and no options bitcoin would build without legacy wallet support if BDB wasn't enabled? And after this change it still does that. Or that before this change plain `./configure` errored if it couldn't find BDB and still does that?

Reply:
This doesn't change default behavior. This only effects a new behavior. The only thing this effects is the case where the user does `./configure --enable-wallet --without-bdb --without-sqlite`. With this change, we throw the error `wallet functionality requestsed but no BDB or SQLite support available`. Without this change, we would compile without wallet support.

As before, if BDB is not found, configure would error.

Reply:
Thanks. Apparently this works because `bitcoin_find_bdb48.m4` doesn't understand what "auto" means. It treats everything that isn't "no" as "yes"